src/Catalog/Helpers/DeletionAuditEntry.cs

```diff
@@ -163,7 +173,7 @@ private void InitValues()
 
             return
                 (await Task.WhenAll(
-                    auditRecords.Select(record => DeletionAuditEntry.Create(auditingStorage, record.Uri, cancellationToken, logger))))
+                    auditRecords.Select(record => DeletionAuditEntry.CreateAsync(auditingStorage, record.Uri, cancellationToken, logger))))
                 // Filter out null records.
```

Comment:
If one Create Async will throw the correct CreatedAsync will be lost. I am not sure if this will impact in any way the scenario but thought to mention it.